src/lobby/react.js

```diff
@@ -286,17 +205,110 @@ class Lobby extends React.Component {
       credentials: this.connection.playerCredentials,
     };
 
-    this.setState({
-      phase: 'play',
-      runningGame: game,
-    });
-  }
+    this.setState({ phase: LobbyPhases.PLAY, runningGame: game });
+  };
 
-  _exitGame() {
-    this.setState({
-      phase: 'list',
-      runningGame: null,
+  _exitRoom = () => {
+    this.setState({ phase: LobbyPhases.LIST, runningGame: null });
+  };
+
+  _getPhaseVisibility = phase => {
+    return this.state.phase !== phase ? 'hidden' : 'phase';
+  };
+
+  renderRooms = (gameInstances, playerName) => {
+    return gameInstances.map(gameInstance => {
+      const { gameID, gameName, players } = gameInstance;
+      return (
+        <LobbyRoomInstance
+          key={'instance-' + gameID}
+          gameInstance={{ gameID, gameName, players: Object.values(players) }}
+          playerName={playerName}
+          onClickJoin={this._joinRoom}
+          onClickLeave={this._leaveRoom}
+          onClickPlay={this._startGame}
+        />
+      );
     });
+  };
+
+  render() {
```

Comment:
Most changes are here:
* `render` method was moved to the end of the class
* `renderer` prop was added here
* rooms rendering was simplified.